visual-js/visual-nightwatch/src/nightwatch/commands/sauceVisualCheck.ts

```diff
@@ -65,12 +74,49 @@ export default class SauceVisualCheck implements NightwatchCustomCommandsModel {
       '';
 
     // Ignore magic
-    const ignoreOptions = options.ignore ?? [];
-    const resolvedIgnoreOptions = parseIgnoreOptions(ignoreOptions);
-    const ignoreRegions =
-      ignoreOptions.length > 0
-        ? await toIgnoreRegionIn(resolvedIgnoreOptions)
-        : [];
+    const isElement = (element: unknown): element is WebElement =>
+      element instanceof WebElement;
+    const getElementMeta = async (element: WebElement) => ({
+      id: await element.getId(),
+    });
+
+    type ParsedElementType = WebElement | WebElement[] | RegionIn;
+
+    const ignorables: RegionType<ParsedElementType>[] = await Promise.all(
+      (options.ignore ?? [])
+        .map<RegionType<ElementType>>((ignore) => {
+          // Resolve strings into ignorables to merge types into a single awaitable type, and give
+          // it a region shape
+          const element =
+            typeof ignore === 'string'
+              ? this.api.element.findAll(ignore)
+              : ignore;
+          return {
+            element,
+          };
+        })
+        // Add our ignore regions of the same shape
+        .concat(options.regions ?? [])
+        .map(async (region) => {
+          const { element, ...other } = region;
+          const awaited = await element;
+          return {
+            ...other,
+            // If the items are an array, await all of them to get the resolved WebElement promises instead.
+            element: Array.isArray(awaited)
+              ? await Promise.all(awaited)
+              : awaited,
+          } satisfies RegionType<ParsedElementType>;
+        }),
+    );
```

Comment:
Why not throw everything into `parseRegionsForAPI` without so much pre-processing and extract the elements in `getElementMeta`? It needs to return an array (see comment below) and that this code should be much simpler.

Reply:
_Some_ preprocessing is going to be required since we have some tooling that currently supports selectors, like nightwatch, but isn't supported in WDIO. But sure we can simplify those and potentially do more insid parseRegionsForAPI.